### PR TITLE
SAMZA-1494: Flush operator state at end of stream

### PR DESCRIPTION
**Pull request by vjagadish1989**
- Propagate operator messages at endOfStream to all down-stream operators.
- Emit all pending windows when endOfStream is reached.
- Flush all state on endOfStream irrespective of auto-commit behavior.

### CODE REVIEW

**Comment by vjagadish1989** [comment]
@xinyuiscool @prateekm @nickpan47

File: samza-core/src/main/java/org/apache/samza/operators/impl/OperatorImpl.java [hunk 1]
```diff
@@ -292,8 +298,8 @@ private final void onEndOfStream(MessageCollector collector, TaskCoordinator coo
    * @param collector message collector
    * @param coordinator task coordinator
    */
-  protected void handleEndOfStream(MessageCollector collector, TaskCoordinator coordinator) {
-    //Do nothing by default
+  protected Collection<RM> handleEndOfStream(MessageCollector collector, TaskCoordinator coordinator) {
```
Comment (xinyuiscool) [inline samza-core/src/main/java/org/apache/samza/operators/impl/OperatorImpl.java]: please update the javadoc and add the @return.

**Comment by prateekm** [comment]
@vjagadish1989 
```[ant:checkstyle] /home/travis/build/apache/samza/samza-core/src/test/java/org/apache/samza/operators/impl/TestWindowOperator.java:50:8: Unused import - org.apache.samza.system.EndOfStreamMessage.
 FAILED
FAILURE: Build failed with an exception.
```

**Comment by vjagadish1989** [comment]
@prateekm Thanks for pointing out the check-style formatting. Fixed!

**Pull request closed without merging**